specification/workloads/resource-manager/Microsoft.Workloads/connectors/preview/2023-10-01-preview/connectors.json

```diff
@@ -0,0 +1,1098 @@
+        },
+        "hdbuserstoreKeyName": {
+          "type": "string",
+          "description": "Name of the HANA Database User Store Key."
+        },
+        "instanceNumber": {
+          "description": "Gets or sets the database instance number.",
+          "type": "string"
+        },
+        "dbInstanceSnapshotBackupPolicy": {
+          "$ref": "#/definitions/DBBackupPolicyProperties"
+        }
+      }
+    },
+    "SqlBackupData": {
+      "description": "Defines the SQL Backup data for a virtual instance for SAP.",
+      "x-ms-discriminator-value": "SQL",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupData"
+        }
+      ],
+      "required": [
+        "backupPolicy"
+      ],
+      "properties": {
+        "backupPolicy": {
+          "$ref": "#/definitions/DBBackupPolicyProperties"
+        }
+      }
+    },
+    "ACSSBackupConnectionProperties": {
+      "description": "Defines the backup connection properties for a virtual instance for SAP.",
+      "type": "object",
+      "properties": {
+        "backupData": {
+          "$ref": "#/definitions/BackupData"
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ConnectorProvisioningState",
+          "readOnly": true
+        },
+        "errors": {
+          "description": "Defines the errors related to backup connection resource of virtual instance for SAP.",
+          "readOnly": true,
+          "$ref": "#/definitions/ConnectorErrorDefinition"
+        }
+      }
+    },
+    "BackupData": {
+      "description": "Information about the recovery services vault and backup policy used for backup.",
+      "type": "object",
+      "required": [
+        "backupType",
+        "recoveryServicesVault"
+      ],
+      "properties": {
+        "backupType": {
+          "$ref": "#/definitions/BackupType"
+        },
+        "recoveryServicesVault": {
+          "$ref": "#/definitions/RecoveryServicesVaultProperties"
+        }
+      },
+      "discriminator": "backupType"
+    },
+    "BackupType": {
+      "description": "The type of backup, VM, SQL or HANA.",
+      "type": "string",
+      "enum": [
+        "VM",
+        "SQL",
+        "HANA"
+      ],
+      "x-ms-mutability": [
+        "create",
+        "read"
+      ],
+      "x-ms-enum": {
+        "name": "backupType",
+        "modelAsString": true
+      }
+    },
+    "ACSSBackupConnection": {
+      "description": "Define the backup connection resource of virtual instance for SAP..",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ACSSBackupConnectionProperties"
+        }
+      }
+    },
+    "ConnectorErrorDefinition": {
```

Comment:
>ConnectorErrorDefinition [](http://example.com/codeflow?start=5&length=24)

Could you please review the ErrorDetail type in common types? I wonder if that is suitable here.

Reply:
This was custom defined during the inception of the service to limit the properties required to be populated and to enhance it as required in future.